tap_square/client.py

```diff
@@ -295,3 +302,46 @@ def get_payments(self, start_time, bookmarked_cursor):
                 raise Exception(result.errors)
 
             yield (result.body.get('payments', []), result.body.get('cursor'))
+
+
+    def get_batch_token(self, link): #pylint: disable=no-self-use
+        if link:
+            url = link[link.find('<')+1:link.find('>')]
+            parsed = urlparse.urlparse(url)
+            batch_token = parse_qs(parsed.query)['batch_token'][0]
+            return int(batch_token)
+        return None
+
+    def get_roles(self, bookmarked_cursor):
+        headers = {
+            'Authorization': 'Bearer ' + self._access_token,
+            'Content-Type': 'application/json'
+        }
+        params = {}
+        url = 'https://connect.squareup.com/v1/me/roles'
+
+
+        if bookmarked_cursor:
+            params['batch_token'] = bookmarked_cursor
+
+        with singer.http_request_timer('GET payments'):
+            result = requests.get(url, headers=headers, params=params)
+
+        # if result.is_error():
+        #     raise Exception(result.errors)
+
+        batch_token = self.get_batch_token(result.headers.get('Link'))
+
+        yield (result.json(), batch_token)
+
+        while batch_token:
+            params['batch_token'] = batch_token
+            with singer.http_request_timer('GET payments'):
+                result = requests.get(url, headers=headers, params=params)
+
+            # if result.is_error():
+            #     raise Exception(result.errors)
```

Comment:
Please uncomment this. All requests should check for errors and throw exception. Otherwise it hides failures.